nemo_skills/training/start_grpo.py

```diff
@@ -0,0 +1,210 @@
+            ptl_model.tokenizer,
+            cfg.model.data.task_to_prompt_key,
+            cfg.model.data.apply_chat_template,
+            cfg.model.data.system_prompt_file,
+            cfg.model.data.prompt_file,
+            seq_length=cfg.model.data.seq_length,
+        )
+        for split in ("train", "validation")
+    ]
+
+    max_seqlen = cfg.model.grpo.length_params.max_length
+    eos_id = ptl_model.tokenizer.eos_id
+
+    # collate fn to pad to the max seq length in the batch
+    collate_fn = environment_collate_with_pad_to_max_batch(
+        max_seqlen, eos_id, cfg, generate_masks_and_position_ids=False
+    )
+
+    train_dataloader_builder = partial(
+        build_dataloader,
+        cfg=cfg,
+        dataset=train_ds,
+        mbs=cfg.trainer.grpo.prompt_micro_batch_size,
+        gbs=cfg.trainer.grpo.num_prompts_per_grpo_step,
+        collate_fn=collate_fn,
+        load_gbs=False,
+        use_random_sampler=cfg.model.data.shuffle_train_data,
+    )
+
+    val_dataloader_builder = partial(
+        build_dataloader,
+        cfg=cfg,
+        dataset=validation_ds,
+        mbs=cfg.trainer.grpo.val_prompt_micro_batch_size,
+        gbs=cfg.trainer.grpo.val_num_prompts_per_grpo_step,
+        collate_fn=collate_fn,
+        load_gbs=False,
+        use_random_sampler=False,
+    )
+
+    # nemo uses the train dataloader to figure out
+    # max steps to take when max_steps = -1
+    # but our train dataloader is for the prompts
+    # so we instaniate a dummy dataloader
+    # to get the proper max *optimization* steps
+    # nemo treats batch size of normal dataloader as GBS/DP
+    # so we need to offset it by DP
+    dummy_train_dataloader = torch.utils.data.DataLoader(
+        dataset=train_ds,
+        batch_size=divide(cfg.trainer.grpo.num_prompts_per_grpo_step, parallel_state.get_data_parallel_world_size()),
+    )
+
+    init_using_ptl(trainer, ptl_model, dummy_train_dataloader, train_ds)
+    # make sure the dummy train dataloader is never used
+    del ptl_model._train_dl
+    del dummy_train_dataloader
+
+    optimizer, scheduler = extract_optimizer_scheduler_from_ptl_model(ptl_model)
+    ckpt_callback = add_custom_checkpoint_callback(trainer, ptl_model)
+
+    logger.log_hyperparams(OmegaConf.to_container(cfg))
+
+    # init environments and rollout generator
+    math_environment = MathEnvironment(cfg.trainer.grpo.environments.math)
+    # your_environment = Environment(cfg)
+    tasks_to_environments = {
+        "math": math_environment,
+        # "your_task": your_environment,
+    }
+    rollout_generator = SequenceRewardRolloutGenerator(cfg.trainer.grpo, tasks_to_environments)
+
+    timer = Timer(cfg.exp_manager.get("max_time_per_run") if cfg.exp_manager else None)
+
+    batch_iterator_cfg = cfg.trainer.grpo.get("batch_iterator", {})
+    batch_iterator_cls = get_batch_iterator_cls(batch_iterator_cfg)
+
+    print([(name, p.shape) for name, p in ptl_model.model.named_parameters()], flush=True)
```

Comment:
Use logger

Reply:
this is a copy-paste from nemo-aligner, so easier not to modify it unless necessary to make it simpler to upgrade in the future